bcc/module.go

```diff
@@ -201,6 +229,74 @@ func (bpf *Module) AttachKretprobe(fnName string, fd int) error {
 	return bpf.attachProbe(evName, BPF_PROBE_RETURN, fnName, fd)
 }
 
+// AttachUprobe attaches a uprobe fd to the symbol in the library or binary 'name'
+// The 'name' argument can be given as either a full library path (/usr/lib/..),
+// a library without the lib prefix, or as a binary with full path (/bin/bash)
+// A pid can be given to attach to, or -1 to attach to all processes
+func (bpf *Module) AttachUprobe(name, symbol string, fd, pid int) error {
+	path, addr, err := resolveSymbolPath(name, symbol, 0x0, 0)
```

Comment:
Yes, you are absolutely correct - AttachUProbe doesn't work when the traced process is running in a different namespace.

But I'm not entirely sure on the reason.

The call to bcc_resolve_symname() (_with_ the pid) _appears_ to come back correctly, with what looks like the correct address (its a little hard to tell as I'm using an Ubuntu container and the address is the same in the parent and child namespace).

However, the trace itself simply doesn't work, no events are triggered.

I looked through the bcc code and failed to find where it does the namespace switching, would you be able to point me towards it?

If its OK with you I'll add a comment and then raise a follow up issue once the PR is merged?

I have amended the code to actually pass in the pid anyway, as it should.